src/data/dataTemplates.js

```diff
@@ -0,0 +1,13 @@
+import PropTypes from "prop-types";
+import {RANKS} from "../util/constants";
+
+export const unitTemplate = {
+    name: PropTypes.string.isRequired,
+    point_cost: PropTypes.number.isRequired,
+    rank: PropTypes.oneOf(Object.values(RANKS)),
+    faction: PropTypes.string.isRequired,
+    type: PropTypes.string,
+    keywords: PropTypes.string,
+    upgrade_slots: PropTypes.arrayOf(PropTypes.string),
+    restricted_upgrades: PropTypes.array
```

Comment:
Are there really any non-required props here? Rank seems required, type seems required, keywords is always there even if it's empty, and restricted upgrades is always there even if it's just an empty list ya?

Reply:
The actual data is very much incomplete and placeholder at the moment, mostly wanted to see if something like this could work, then will go back and change it all.